cpp/src/link_prediction/similarity_impl.cuh

```diff
@@ -58,7 +59,62 @@ rmm::device_uvector<weight_t> similarity(
     //        max weight((u,a), (a,v)).
     //    Use these to compute weighted score
     //
-    CUGRAPH_FAIL("weighted similarity computations are not supported in this release");
+
+    rmm::device_uvector<weight_t> similarity_score(num_vertex_pairs, handle.get_stream());
+
+    //
+    //  Compute vertex_degree for all vertices, then distribute to each GPU.
+    //  Need to use this instead of the dummy properties below
+    //
+
+    rmm::device_uvector<weight_t> weighted_out_degrees =
+      compute_out_weight_sums(handle, graph_view, *edge_weight_view);
+
+    rmm::device_uvector<weight_t> vertex_weights =
+      compute_out_weight_sums(handle, graph_view, *edge_weight_view);
+
+    per_v_pair_transform_dst_nbr_intersection(
+      handle,
+      graph_view,
+      *edge_weight_view,
+      vertex_pairs_begin,
+      vertex_pairs_begin + num_vertex_pairs,
+      weighted_out_degrees.begin(),
+      [functor] __device__(auto a,
+                           auto b,
+                           auto weight_a,
+                           auto weight_b,
+                           auto intersection,
+                           auto intersected_properties_a,
+                           auto intersected_properties_b) {
+        weight_t min_weight_a_intersect_b = weight_t{0};
+        weight_t max_weight_a_intersect_b = weight_t{0};
+        weight_t sum_of_intersected_a     = weight_t{0};
+        weight_t sum_of_intersected_b     = weight_t{0};
+
+        for (size_t k = 0; k < intersection.size(); k++) {
```

Comment:
Does this work properly across multiple GPUs?  If we do 2D partitioning of edges, the outgoing edges for vertex `v` might be spread across a row of GPUs in the 2D configuration.

In this case, for a vertex `v` that has edges spread across multiple GPus, to compute the min/max and sums required, we would need to execute this functor and then do a min reduction, a max reduction, and 2 sum reductions across the GPUs to have the proper value.

I don't see anything that would do this step.

Reply:
Just to clarify, you are asserting the following:  if vertex `u` and vertex `v` have 1000 vertices in the neighbor intersection (and those edges are stored across multiple GPUs), then the edge information for all 1000 vertices is consolidated onto a single GPU in this data structure to perform the computation.

I haven't dug into the code, so I would rely on @seunghwak for this as well.